src/libraries/Microsoft.PowerFx.Connectors/PowerPlatformConnectorClient.cs

```diff
@@ -42,23 +44,90 @@ public class PowerPlatformConnectorClient : HttpClient
+        /// <param name="userAgent">Product UserAgent to add to Power-Fx one (Power-Fx/version).</param>
+        /// <param name="httpInvoker">Optional HttpMessageInvoker. If not provided a default HttpClient is used.</param>
+        public PowerPlatformConnectorClient(OpenApiDocument swaggerFile, string environmentId, string connectionId, Func<string> getAuthToken, string userAgent, HttpMessageInvoker httpInvoker = null)
+            : this(swaggerFile.GetAuthority() ?? throw new ArgumentException("Swagger document doesn't contain an endpoint"), environmentId, connectionId, getAuthToken, userAgent, httpInvoker)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PowerPlatformConnectorClient"/> class.        
+        /// </summary>
+        /// <param name="endpoint">APIM Endpoint.</param>
+        /// <param name="environmentId">Environment Id.</param>
+        /// <param name="connectionId">Connection/connector Id.</param>
+        /// <param name="getAuthToken">Function returning the JWT token.</param>
+        /// /// <param name="userAgent">Product UserAgent to add to Power-Fx one (Power-Fx/version).</param>
+        /// <param name="httpInvoker">Optional HttpMessageInvoker. If not provided a default HttpClient is used.</param>
+        public PowerPlatformConnectorClient(string endpoint, string environmentId, string connectionId, Func<string> getAuthToken, string userAgent, HttpMessageInvoker httpInvoker = null)
+            : this(endpoint, environmentId, connectionId, async () => getAuthToken(), userAgent, httpInvoker)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PowerPlatformConnectorClient"/> class.        
+        /// </summary>
+        /// <param name="endpoint">APIM Endpoint.</param>
+        /// <param name="environmentId">Environment Id.</param>
+        /// <param name="connectionId">Connection/connector Id.</param>
+        /// <param name="getAuthToken">Async function returning the JWT token.</param>
+        /// <param name="userAgent">Product UserAgent to add to Power-Fx one (Power-Fx/version).</param>
+        /// <param name="httpInvoker">Optional HttpMessageInvoker. If not provided a default HttpClient is used.</param>
+        public PowerPlatformConnectorClient(string endpoint, string environmentId, string connectionId, Func<Task<string>> getAuthToken, string userAgent, HttpMessageInvoker httpInvoker = null)
         {
             _client = httpInvoker ?? new HttpClient();
 
             GetAuthToken = getAuthToken ?? throw new ArgumentNullException(nameof(getAuthToken));
             ConnectionId = connectionId ?? throw new ArgumentNullException(nameof(connectionId));
             EnvironmentId = environmentId ?? throw new ArgumentNullException(nameof(environmentId));
+            UserAgent = string.IsNullOrWhiteSpace(userAgent) ? $"PowerFx/{Version}" : $"{userAgent} PowerFx/{Version}";
```

Comment:
Just double checking if this is the scheme (with a space). 
Separating by space seems weird.

Reply:
We follow RFC9110 here and spaces are OK

User-Agent = product *( RWS ( product / comment ) )
RWS = 1*( SP / HTAB )